File: 2561-number-of-distinct-averages/number-of-distinct-averages.cpp
class Solution {
public:
    int distinctAverages(vector<int>& nums) {
        set<double>set;
        int n=nums.size();
        sort(nums.begin(),nums.end());
        int i=0,j=n-1;
        while(i<j){
            double avg=(nums[i]+nums[j])/2.0;
            set.insert(avg);
            j--;
            i++;
        }
        return set.size();
    }
};

File: 2561-number-of-distinct-averages/number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        Set<Double> set=new HashSet<>();
        Arrays.sort(nums);
        int i=0,j=nums.length-1;
        while(i<j){
            double avg=(nums[i]+nums[j])/2.0;
            set.add(avg);
            i++;
            j--;
        }
        
        return set.size();
    }
}